### PR TITLE
dta: initial Android support for the env.json file

### PR DESCRIPTION
**Pull request by toddsouthenbentley**

### CODE REVIEW

File: test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt [hunk 1]
```diff
@@ -115,20 +115,42 @@ class AssetsPathHandler(context: Context) : WebViewAssetLoader.PathHandler {
...
         host = IModelJsHost(this, alwaysExtractAssets, true)
         host.startup()
-        val frontend = object : MobileFrontend(host, "&standalone=true") {
+
+        var args = "&standalone=true"
+        loadEnvJson()
+        env.optString("IMJS_STANDALONE_FILENAME", "").takeIf { it.isNotEmpty() }?.let { fileName ->
+            // ensure fileName already exists in the external files
+            getExternalFilesDir(BIM_CACHE_DIR)?.let { filesDir ->
+                File(filesDir, fileName).takeIf { it.exists() }?.let {
```
Comment (tcobbs-bentley) [inline test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt:140]: In my opinion, having two it variables so close together with completely different meanings is confusing. I would suggest using `bimFile -> bimFile.exists()` here, and `bimFilename -> bimFileName.isNotEmpty()` above.

Comment (tcobbs-bentley) [reply test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt:140]: Oh, and `bimFile ->` at the end of the line?

Comment (tcobbs-bentley) [reply test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt:140]: (The fact taht I'm not even sure if the `it` on the following line (141) is a File hopefully adds to my case for this being confusing.

File: test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt [hunk 2]
```diff
@@ -115,20 +115,42 @@ class AssetsPathHandler(context: Context) : WebViewAssetLoader.PathHandler {
...
+
+        if (env.has("IMJS_IGNORE_CACHE"))
+            args += "&ignoreCache=true"
+
+        val frontend = object : MobileFrontend(host, args) {
             override fun supplyEntryPoint(): String {
-                // If you want to connect to a local dev server instead of the built-in frontend, return something like: "192.168.86.20:3000"
-                return "https://${WebViewAssetLoader.DEFAULT_DOMAIN}/assets/frontend/index.html"
+                // Connect to a local dev server if specified in the env JSON, otherwise use the embedded frontend
+                return env.optString("IMJS_DEBUG_URL", "").takeIf { it.isNotEmpty() } ?:
```
Comment (tcobbs-bentley) [inline test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt:152]: I wonder if optNonEmptyString should be added as an extension.

File: test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt (hunk 1 above)
Comment (toddsouthenbentley) [reply test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt]: I refactored the code to be more readable.

File: test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt (hunk 2 above)
Comment (toddsouthenbentley) [reply test-apps/display-test-app/android/imodeljs-test-app/app/src/main/java/com/bentley/imodeljs_test_app/MainActivity.kt]: Done

**Review by tcobbs-bentley — APPROVED** [review]

**Pull request merged**